src/common/ConfirmationDialogue.tsx

```diff
@@ -0,0 +1,45 @@
+import Button from '@mui/material/Button'
+import Dialog from '@mui/material/Dialog'
+import DialogTitle from '@mui/material/DialogTitle'
+import DialogActions from '@mui/material/DialogActions'
+import Typography from '@mui/material/Typography'
+import DialogContent from '@mui/material/DialogContent'
+import Alert from '@mui/material/Alert'
+
+type ConfirmationDialogProps = {
+  open: boolean
+  title: string
+  onCancel: () => void
+  onConfirm: () => void
+  errorMessage?: string
+}
+
+export default function ConfirmationDialogue({
+  open,
+  title,
+  onCancel,
+  onConfirm,
+  errorMessage,
```

Comment:
As this is optional we could do `errorMessage = '',` so we don't have to handle `undefined`